### PR TITLE
skeleton: integrate the latest stable wolfssl with DCAP support

### PR DESCRIPTION
**Pull request by haosanzi**
Fixed: #538
Signed-off-by: Shirong Hao <shirong@linux.alibaba.com>

### CODE REVIEW

File: ra-tls/ra-tls-server/lib/Makefile [hunk 1]
```diff
@@ -83,7 +83,7 @@ ifndef ECDSA
 endif
 
 App_Include_Paths := $(Wolfssl_Include_Paths) -I$(SGX_SDK)/include -I$(SGX_RA_TLS_ROOT) -I$(INCDIR) $(SGX_DCAP_INC) -I$(shell readlink -f .)
-App_C_Flags := $(SGX_COMMON_CFLAGS) -fPIE -Wno-attributes -Wall -Wno-unused-const-variable $(App_Include_Paths) $(Wolfssl_C_Extra_Flags)
+App_C_Flags := $(SGX_COMMON_CFLAGS) -fPIC -Wno-attributes -Wall -Wno-unused-const-variable $(App_Include_Paths) $(Wolfssl_C_Extra_Flags)
```
Comment (jiazhang0) [inline ra-tls/ra-tls-server/lib/Makefile:86]: So what happens if -fPIE is used?

File: rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md [hunk 2]
```diff
@@ -71,6 +66,42 @@ Then build the skeleton docker image with the command:
...
+
+COPY Wolfssl_Enclave.signed.so /
+
+RUN mkdir -p /etc/ssl/certs/
+COPY ca-bundle.trust.crt /etc/ssl/certs/
+
+COPY sgx_default_qcnl.conf /etc/
+
+COPY libsgx_pce.signed.so /usr/lib64
+COPY libsgx_qe3.signed.so /usr/lib64
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md:95]: This assumes PCE and QE3 both exist in container. It is OK but not the goal. Currently intelsgx has implemented protobuf message used to communicate with aesmd, so it is worthy doing better here in future. I mean PCE and QE3 are hosted by aesmd.

File: rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md [hunk 3]
```diff
@@ -71,6 +66,42 @@ Then build the skeleton docker image with the command:
...
+
+cat >Dockerfile <<EOF
+FROM centos:8.1.1911
+
+WORKDIR /
+
+COPY Wolfssl_Enclave.signed.so /
+
+RUN mkdir -p /etc/ssl/certs/
+COPY ca-bundle.trust.crt /etc/ssl/certs/
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md:90]: Why doing this? The trusted certificate is used to authenticate what?

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

File: ra-tls/ra-tls-server/lib/Makefile (hunk 1 above)
Comment (haosanzi) [reply ra-tls/ra-tls-server/lib/Makefile:86]: Build `libskeleton-v3.so`  will fail with the information. `relocation xxx can not be used when making a shared object; recompile with --FPIC`

File: rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md (hunk 3 above)
Comment (haosanzi) [reply rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md]: There is no need to add `ca-bundle.trust.crt` in `dcap`  remote attestation. I have removed it.

File: ra-tls/ra-tls-server/lib/Makefile (hunk 1 above)
Comment (jiazhang0) [reply ra-tls/ra-tls-server/lib/Makefile:86]: Please submit an issue and set its fix milestone to 0.7.0

File: rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md (hunk 2 above)
Comment (jiazhang0) [reply rune/libenclave/internal/runtime/pal/skeleton/running_skeleton_with_tls_server.md:92]: Please submit an issue to address the above issue and set its fix milestone to 0.7.0

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

**Review by jiazhang0 — APPROVED** [review]

**Review by jiazhang0 — APPROVED** [review]

**Pull request merged**